### PR TITLE
min-11 && min-11.1

### LINKED ISSUE #13: Analytics Package
App should have analytics integration

### LINKED ISSUE #63: 11.1 `nexa doctor` Cleaner Scripts
# Using `nexa doctor` 
- Execute this command from the root directory first to address any troubles within the monorepo.
- If this does not help, reach out for help

### PR DESCRIPTION
**Pull request by deontae-smith**
# min-11 Nexa Development Environment

## tl;dr
- The Nexa development environment is a set of DevOps scripts designed to make the developer engineering process better.
- **While these scripts mainly consist of shell commands, there are a few extra steps required after this PR. Please review the 'Next Steps' section of this PR for specific instructions, and follow them in your terminal to ensure smooth implementation**

# min-11.1 Nexa doctor Cleaner Scripts

## tl;dr 
- In a monorepo, developers may encounter issues such as bugs, slow performance, or errors while working on Nexa. These problems typically stem from dependencies or peer dependencies rather than the setup itself.
- This PR introduces a new feature: the ability for developers to run the command `nexa doctor` 
- Executing this command will automatically address many common issues.
- If you receiving a success message after running `nexa doctor` attempt to run `pnpm dev` again.

### Common Questions/Concerns :
- Should the doctor process be fast?
-- The doctor process may be slower for some and faster then others depending on the issue. 

- Files are turning red or breaking when going to the doctor
-- Some files will break as the doctor process runs, after the a successful visit files are no longer red. 

# Next Steps (Important)
Please follow these steps to ensure a successful process 

## First
- You will need to inject your terminal, but its not as scary as it sounds.
- Before you start note this opreation is based on `zsh` shell terminal. You can check this by looking at your terminal in vscode. 

Example Photo:
Screenshot 2024-02-11 at 6 28 02 PM
**If your terminal is not zsh, stop and contact @deontae-smith**

## Second 
Ensure you are on the main branch, and the latest changes are pulled 
(This includes the changes from this PR)
- In the terminal run `nano ~/.zshrc`       
You should see the terminal output open and you will be allowed to paste output.
- Copy and paste the code below into the terminal

```
nexa() {
    local script_path="./nexa/$1"
    if [[ -f "$script_path" && -x "$script_path" ]]; then
        shift
        "$script_path" "$@"
    else
        echo -e "\e[91;1mNexa not found, are you running from the root?\e[0m"
    fi
}
```

**If successful your terminal will look something like the photo below.**
 
Screenshot 2024-02-11 at 6 33 17 PM

- Save the terminal with ctrl+X and type Y to confirm
- run `chmod +x ./nexa/*`

- After this you should be good. When bugs come remember to try  `nexa doctor` first. 
(nexa doctor must be ran from the root, otherwise a red error message will appear.) 

Closes #13 (Parent)
Closes #63 (Task)

### CODE REVIEW

**Comment by deontae-smith** [comment]
STALE
